File: app/src/main/java/me/zipi/navitotesla/ui/setting/ConditionRecyclerAdapter.java
package me.zipi.navitotesla.ui.setting;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import java.lang.ref.WeakReference;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import me.zipi.navitotesla.R;

public class ConditionRecyclerAdapter extends RecyclerView.Adapter<ConditionRecyclerAdapter.ViewHolder> {

    List<String> items;
    final OnDeleteButtonClicked listener;

    public ConditionRecyclerAdapter(OnDeleteButtonClicked listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.condition_view, parent, false);
        return new ViewHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.onBind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setItems(List<String> items) {
        this.items = items;
        this.notifyDataSetChanged();
    }

    public interface OnDeleteButtonClicked {
        void onClick(int position);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        final TextView nameView;
        final Button deleteButton;
        private final WeakReference<OnDeleteButtonClicked> listenerRef;

        public ViewHolder(@NonNull View itemView, OnDeleteButtonClicked listener) {
            super(itemView);

            nameView = itemView.findViewById(R.id.recylcer_bluetooth);
            deleteButton = itemView.findViewById(R.id.btnConditionDelete);
            listenerRef = new WeakReference<>(listener);
            deleteButton.setOnClickListener(this);
        }

        void onBind(String item) {
            if (item != null) {
                nameView.setText(item);
            }
        }

        @Override
        public void onClick(View view) {
            if (listenerRef != null && listenerRef.get() != null) {
                listenerRef.get().onClick(getAdapterPosition());
            }
        }
    }

}

File: app/src/main/kotlin/me/zipi/navitotesla/ui/setting/ConditionRecyclerAdapter.kt
package me.zipi.navitotesla.ui.setting

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import me.zipi.navitotesla.R
import java.lang.ref.WeakReference

class ConditionRecyclerAdapter(
    private val listener: OnDeleteButtonClicked,
) : RecyclerView.Adapter<ConditionRecyclerAdapter.ViewHolder>() {
    private var items: List<String> = mutableListOf()

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int,
    ): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.condition_view, parent, false)
        return ViewHolder(view, listener)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int,
    ) {
        holder.onBind(items[position])
    }

    override fun getItemCount(): Int = items.size

    @SuppressLint("NotifyDataSetChanged")
    fun setItems(items: List<String>) {
        this.items = items
        notifyDataSetChanged()
    }

    interface OnDeleteButtonClicked {
        fun onClick(position: Int)
    }

    class ViewHolder(
        itemView: View,
        listener: OnDeleteButtonClicked,
    ) : RecyclerView.ViewHolder(itemView),
        View.OnClickListener {
        private val nameView: TextView
        private val deleteButton: Button
        private val listenerRef: WeakReference<OnDeleteButtonClicked>

        init {
            nameView = itemView.findViewById(R.id.recylcer_bluetooth)
            deleteButton = itemView.findViewById(R.id.btnConditionDelete)
            listenerRef = WeakReference(listener)
            deleteButton.setOnClickListener(this)
        }

        fun onBind(item: String?) {
            if (item != null) {
                nameView.text = item
            }
        }

        override fun onClick(view: View) {
            listenerRef.get()?.onClick(adapterPosition)
        }
    }
}
